Decode/PerfItemType.cs

```diff
@@ -6,90 +6,151 @@ namespace Microsoft.LinuxTracepoints.Decode
+    /// Otherwise, this is an array of variable-length values, <c>Format</c> is significant,
+    /// and <c>StructFieldCount</c> should be ignored.
     /// </item>
+    ///
     /// </list>
     /// </summary>
-    public readonly ref struct PerfItemType
+    public readonly ref struct PerfItemMetadata
     {
         /// <summary>
-        /// Initializes a new instance of the EventHeaderItemType struct. These are normally created
-        /// by EventHeaderEnumerator.GetItemType().
+        /// Initializes a new instance of the PerfItemMetadata struct.
+        /// <br/>
+        /// These are not normally created directly. You'll normally get instances of this struct from
+        /// <see cref="EventHeaderEnumerator"/><c>.GetItemMetadata()</c> or
+        /// <see cref="PerfFieldFormat"/><c>.GetFieldValue()</c>.
         /// </summary>
-        public PerfItemType(
+        /// <param name="byteReader">
+        /// Reader that is configured for the event data's byte order.
+        /// </param>
+        /// <param name="encodingAndArrayFlag">
+        /// The field encoding, including the appropriate array flag if the field is an array element,
+        /// array-begin, or array-end. The chain flag must be unset.
+        /// </param>
+        /// <param name="format">
+        /// The field format. The chain flag must be unset.
+        /// </param>
+        /// <param name="isScalar">
+        /// True if this represents a non-array value or a single element of an array.
+        /// False if this represents an array-begin or an array-end.
+        /// </param>
+        /// <param name="typeSize">
+        /// For simple encodings (e.g. Value8, Value16, Value32, Value64, Value128),
+        /// this is the size of one element in bytes (1, 2, 4, 8, 16). For complex types
+        /// (e.g. Struct or string), this is 0.
+        /// </param>
+        /// <param name="elementCount">
+        /// For array-begin or array-end, this is number of elements in the array.
+        /// For non-array or for array element, this is 1.
+        /// This may be 0 in the case of a variable-length array of length 0.
+        /// </param>
+        /// <param name="fieldTag">
+        /// Field tag, or 0 if none.
+        /// </param>
+        public PerfItemMetadata(
             PerfByteReader byteReader,
-            EventHeaderFieldEncoding encodingAndArrayFlags,
+            EventHeaderFieldEncoding encodingAndArrayFlag,
             EventHeaderFieldFormat format,
+            bool isScalar,
             byte typeSize,
             ushort elementCount,
             ushort fieldTag = 0)
         {
             this.ElementCount = elementCount;
             this.FieldTag = fieldTag;
             this.TypeSize = typeSize;
-            this.EncodingAndArrayFlags = encodingAndArrayFlags;
+            this.EncodingAndArrayFlagAndIsScalar = encodingAndArrayFlag | (isScalar ? EventHeaderFieldEncoding.ChainFlag : 0);
```

Comment:
Note: To keep this struct at its current size of 8 bytes, stash the IsScalar value into the already-overloaded `EncodingAndArrayFlags` property.